api/src/integration/java/org/pmiops/workbench/google/DirectoryServiceImplIntegrationTest.java

```diff
@@ -2,37 +2,23 @@
 
 import static com.google.common.truth.Truth.assertThat;
 
-import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
-import com.google.api.client.http.apache.ApacheHttpTransport;
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
 import java.time.Clock;
 import java.util.Map;
-import org.junit.Before;
 import org.junit.Test;
-import org.pmiops.workbench.config.WorkbenchConfig;
-import org.pmiops.workbench.test.Providers;
+import org.pmiops.workbench.BaseIntegrationTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Import;
 import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
-import org.springframework.retry.backoff.NoBackOffPolicy;
 import org.springframework.retry.policy.SimpleRetryPolicy;
 import org.springframework.retry.support.RetryTemplate;
 
-public class DirectoryServiceImplIntegrationTest {
-  private DirectoryServiceImpl service;
-  private final GoogleCredential googleCredential = getGoogleCredential();
-  private final WorkbenchConfig workbenchConfig = createConfig();
-  private final ApacheHttpTransport httpTransport = new ApacheHttpTransport();
+public class DirectoryServiceImplIntegrationTest extends BaseIntegrationTest {
+  @Autowired private DirectoryService service;
 
-  @Before
-  public void setUp() {
-    service =
-        new DirectoryServiceImpl(
-            Providers.of(googleCredential),
-            Providers.of(workbenchConfig),
-            httpTransport,
-            new GoogleRetryHandler(new NoBackOffPolicy()));
-  }
+  @TestConfiguration
+  @Import(DirectoryServiceImpl.class)
+  static class Configuration {}
 
   @Test
   public void testDummyUsernameIsNotTaken() {
```

Comment:
Wait, so this test is explicitly making sure usernames are recyclable? I didn't know we allowed that. What's the benefit of allowing that?

Reply:
This is not making any statement about recycling usernames. It's testing that a user which has never been created is not taken. The test would break if someone create such a user, but hopefully they won't as indicated by the username.